stdlib/LinearAlgebra/src/triangular.jl

```diff
@@ -2607,7 +2607,11 @@ function sqrt(A::UpperTriangular{T},::Val{realmatrix}) where {T,realmatrix}
             @simd for k = i+1:j-1
                 r -= R[i,k]*R[k,j]
             end
-            iszero(r) || (R[i,j] = sylvester(R[i,i],R[j,j],-r))
+            if iszero(R[i,i])  && iszero(R[j,j])
+                R[i,j] = 0
```

Comment:
Isn't `R[i,j]` already zero?

Reply:
yes you're right, fixed the ifelse in 573d006